statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,726 @@
+        if start is None:
+            start = self._index[-1] + 1
+        start, end, out_of_sample, prediction_index = self._get_prediction_index(
+            start=start, end=end)
+        if out_of_sample > 0:
+            res = self._predict(h=out_of_sample, **params)
+        else:
+            res = self._predict(h=0, **params)
+        return res.fittedfcast[start:end + out_of_sample + 1]
+
+    def fit(self, alpha=None, beta=None, gamma=None, phi=None, optimized=True,
+            use_boxcox=False, remove_bias=False, use_basinhopping=False):
+        """
+        fit Holt Winter's Exponential Smoothing
+
+        Parameters
+        ----------
+        alpha : float, optional
+            The alpha value of the simple exponential smoothing, if the value is
+            set then this value will be used as the value.
+        beta :  float, optional
+            The beta value of the holts trend method, if the value is
+            set then this value will be used as the value.
+        gamma : float, optional
+            The gamma value of the holt winters seasonal method, if the value is
+            set then this value will be used as the value.
+        phi : float, optional
+            The phi value of the damped method, if the value is
+            set then this value will be used as the value.
+        optimized : bool, optional
+            Should the values that have not been set above be optimized automatically?
+        use_boxcox : {True, False, 'log'}, optional
+            Should the boxcox tranform be applied to the data first? If 'log' then
+            apply the log.
+        remove_bias : bool, optional
+            Should the bias be removed from the fcast and fitted values before being
+            returned?
+        use_basinhopping : bool, optional
+            Should the opptimser try harder using basinhopping to find optimal values?
+
+        Returns
+        -------
+        results : HoltWintersResults class
+            See statsmodels.tsa.holtwinters.HoltWintersResults
+
+        Notes
+        -----
+        This is a full implementation of the holt winters exponential smoothing as
+        per [1]. This includes all the unstable methods as well as the stable methods.
+        The implementaion of the library follows as per the R library as much as possible
+        whilest still being pythonic.
+
+        References
+        ----------
+        [1] Hyndman, Rob J., and George Athanasopoulos. Forecasting: principles and practice. OTexts, 2014.
+        """
+        data = self.endog
+        damped = self.damped
+        seasoning = self.seasoning
+        trending = self.trending
+        trend = self.trend
+        seasonal = self.seasonal
+        m = self.m
+        opt = None
+        phi = phi if damped else 1.0
+        n = len(data)
+        if use_boxcox == 'log':
+            lamda = 0.0
+            y = boxcox(data, 0.0)
+        elif use_boxcox:
+            y, lamda = boxcox(data)
```

Comment:
What's the default method for choosing lambda?
This might need a `use_boxcox = a_float` options, so users can choose the lambda

Reply:
done